packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -172,6 +181,14 @@ const ElementHook: Hooks<VElement> = {
         removeNodeHook(vnode, parentNode);
         removeElmHook(vnode);
     },
+    hydrate: (vnode, node) => {
+        vnode.elm = node as Element;
+
+        createElmHook(vnode);
```

Comment:
Running the full `createElmHook` is wasteful here. The server-side rendered content already has attributes in place, props and event listeners are missing.